### PR TITLE
#2126 add SQLLoggerTest

### LINKED ISSUE #2126: SQLLogger should have a junit test
## Feature Request

**For English only**, other languages will not accept.

Please pay attention on issues you submitted, because we maybe need more details. 
If no response **more than 7 days** and we cannot make decision by current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Is your feature request related to a problem?
No.
### Describe the feature you would like.
SQLLogger should have a junit test.

### PR DESCRIPTION
**Pull request by betterjava**
Fixes #2126 .

Changes proposed in this pull request:
- add SQLLogTest to test SQLLogger

### CODE REVIEW

File: sharding-core/sharding-core-route/src/test/java/org/apache/shardingsphere/core/route/SQLLoggerTest.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public final class SQLLoggerTest {
+
+    private String sql;
```
Comment (terrymanu) [inline sharding-core/sharding-core-route/src/test/java/org/apache/shardingsphere/core/route/SQLLoggerTest.java]: Fields should have a blank line each other

File: sharding-core/sharding-core-route/src/test/java/org/apache/shardingsphere/core/route/SQLLoggerTest.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public final class SQLLoggerTest {
```
Comment (terrymanu) [inline sharding-core/sharding-core-route/src/test/java/org/apache/shardingsphere/core/route/SQLLoggerTest.java]: Should use mock, not test real logback

File: sharding-core/sharding-core-route/src/test/java/org/apache/shardingsphere/core/route/SQLLoggerTest.java (hunk 1 above)
Comment (betterjava) [reply sharding-core/sharding-core-route/src/test/java/org/apache/shardingsphere/core/route/SQLLoggerTest.java]: Ok,I will fix this.

File: sharding-core/sharding-core-route/src/test/java/org/apache/shardingsphere/core/route/SQLLoggerTest.java (hunk 2 above)
Comment (betterjava) [reply sharding-core/sharding-core-route/src/test/java/org/apache/shardingsphere/core/route/SQLLoggerTest.java]: Mockito can not support mock static method.  Should I add powermock to the project dependency tree?

Comment (betterjava) [reply sharding-core/sharding-core-route/src/test/java/org/apache/shardingsphere/core/route/SQLLoggerTest.java]: Mockito can not support mock static method.  Should I add powermock to the project dependency tree?

**Comment by terrymanu** [comment]
`log.info()` is not a static method

**Comment by betterjava** [comment]
I can not find a way to mock a logger for SQLLogger. Mockito can create a mock for Logger,but I can not figure out how to replace real logger  with the mock logger.So I close the pull now.

**Comment by betterjava** [comment]
PowerMock can do this work.
```
        Logger log = PowerMockito.mock(Logger.class);
        PowerMockito.mockStatic(LoggerFactory.class);
        PowerMockito.when(LoggerFactory.getLogger("ShardingSphere-SQL")).thenReturn(log);
        SQLLogger.logSQL("select * from user", Arrays.asList("db1", "db2"));
        InOrder inOrder = Mockito.inOrder(log);
        inOrder.verify(log).info("Rule Type: master-slave", new Object[]{});
        inOrder.verify(log).info("SQL: {} ::: DataSources: {}", new Object[]{ "select * from user","db1,db2"});
```

**Comment by terrymanu** [comment]
Import new mock dependency is unnecessary.
method `log.info()` is not static method, why we have to import power mock to mock this?

**Comment by terrymanu** [comment]
You can use reflection to inject mocked logger

**Comment by betterjava** [comment]
Thanks,I will try.

**Pull request closed without merging**